File: rock_paper_scissors.py
def rock_paper_scissors():
    import random

    choices = ["rock", "paper", "scissors"]

    while True:
        user_choice = input("Choose: rock, paper, or scissors (or q to quit): ").lower()
        if user_choice == 'q':
            break
        elif user_choice not in choices:
            print("Invalid choice. Choose again.")
            continue

        computer_choice = random.choice(choices)
        print("Computer chose:", computer_choice)

        if user_choice == computer_choice:
            print("It's a tie!")
        elif (user_choice == "rock" and computer_choice == "scissors") or (user_choice == "scissors" and computer_choice == "paper") or (user_choice == "paper" and computer_choice == "rock"):
            print("You win!")
        else:
            print("You lose!")

rock_paper_scissors()

File: rock_paper_scissors.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int playerChoice, computerChoice;
    srand(time(0));

    printf("Welcome to Rock, Paper, Scissors Game!\n");
    printf("Choose:\n1 for Rock\n2 for Paper\n3 for Scissors\n");

    printf("Enter your choice: ");
    scanf("%d", &playerChoice);

    if (playerChoice < 1 || playerChoice > 3) {
        printf("Invalid choice!");
        return 1;
    }

    computerChoice = rand() % 3 + 1; // Generate a random choice for the computer

    if (playerChoice == computerChoice) {
        printf("It's a tie!\n");
    } else if ((playerChoice == 1 && computerChoice == 3) ||
               (playerChoice == 2 && computerChoice == 1) ||
               (playerChoice == 3 && computerChoice == 2)) {
        printf("You win!\n");
    } else {
        printf("Computer wins!\n");
    }

    return 0;
}
